packages/firebase_database/example/lib/main.dart

```diff
@@ -28,36 +28,47 @@ class MyHomePage extends StatefulWidget {
   _MyHomePageState createState() => new _MyHomePageState();
 }
 
-const String kTestString = "Hello world!";
-
 class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
-  final DatabaseReference _reference = FirebaseDatabase.instance.reference();
-  StreamSubscription _childAdded;
+  int _counter;
+  final DatabaseReference _counterRef = FirebaseDatabase.instance.reference()
+      .child('counter');
+  final DatabaseReference _messagesRef = FirebaseDatabase.instance.reference()
+      .child('messages');
+  StreamSubscription _counterSubscription;
+  StreamSubscription _messagesSubscription;
+  bool _anchorToBottom = false;
 
   String _kTestKey = 'Hello';
   String _kTestValue = 'world!';
 
   @override
   void initState() {
     super.initState();
-    _childAdded = _reference.onChildAdded.listen((Event event) {
-      assert(event.snapshot.value[_kTestKey] == _kTestValue);
-      print("Child added: ${event.snapshot.value}");
+    _counterSubscription = _counterRef.onValue.listen((Event event) {
       setState(() {
-        _counter++;
+        _counter = event.snapshot.value ?? 0;
       });
     });
+    _messagesSubscription = _messagesRef.onChildAdded.listen((Event event) {
+      print("Child added: ${event.snapshot.value}");
```

Comment:
General pedanto-nit, here and elsewhere: looks like there's a mix of single- and double-quoted strings throughout the Dart code, looks like more single though, so maybe stick to those except where there are nested single quotes.